src/main/java/step2/Calculator.java

```diff
@@ -1,56 +1,34 @@
 package step2;
 
+import step2.FourOperation.*;
+
 public class Calculator {
-    private static final String PLUS = "+";
-    private static final String MINUS = "-";
-    private static final String DIVIDE = "/";
-    private static final String MULTIPLY = "*";
 
-    public int operation(String input) {
+    private OperationFactory factory = new FourOperationFactory();
+
+    public int fourRuleCalculations(String input) {
 
         String[] inputArg = input.split(" ");
 
         isEmptyInput(input);
 
         int leftNum = Integer.valueOf(inputArg[0]);
-        for(int i = 1; i < inputArg.length-1; i++) {
-            System.out.println(inputArg[i]);
-            if(inputArg[i].equals(PLUS)){
-                leftNum = plus(leftNum, Integer.valueOf(inputArg[i+1]));
-            }
-            if(inputArg[i].equals(MINUS)){
-                leftNum = minus(leftNum, Integer.valueOf(inputArg[i+1]));
-            }
-            if(inputArg[i].equals(DIVIDE)){
-                leftNum = divide(leftNum, Integer.valueOf(inputArg[i+1]));
-            }
-            if(inputArg[i].equals(MULTIPLY)){
-                leftNum = multiply(leftNum, Integer.valueOf(inputArg[i+1]));
-            }
+        for (int i = 1; i < inputArg.length - 1; i=i+2) {
+            Operation fourRuleOperation = operand(inputArg[i]);
+            leftNum = fourRuleOperation.operation(leftNum, Integer.valueOf(inputArg[i+1]));
         }
         return leftNum;
     }
 
-    public void isEmptyInput(String input) {
-        if(input == null || input.length() == 0)
-            throw new IllegalArgumentException("입력 값이 null 이거나 빈 공백 문자일 경우 에러발생");
-    }
-
-    public int plus(int input1, int input2) {
-        return input1 + input2;
-    }
-
-    public int minus(int input1, int input2) {
-        return input1 - input2;
+    private Operation operand(String operand){
+        return factory.getOperation(operand);
     }
 
-    public int divide(int input1, int input2) {
-        return input1 / input2;
-    }
-
-    public int multiply(int input1, int input2) {
-        return input1 * input2;
+    private void isEmptyInput(String input) {
+        if(input == null || input.length() == 0){
```

Comment:
조건문 사용시 다음과 같이 if 와 괄호 사이에 공백을 넣어주시는게 컨벤션이오니 참고 부탁드려요 :)
[7.4절](https://myeonguni.tistory.com/1596)
https://naver.github.io/hackday-conventions-java/#space-between-keyword-parentheses

```suggestion
        if (input == null || input.length() == 0) {
```